cnd/src/swap_protocols/rfc003/alice/actions/erc20.rs

```diff
@@ -17,15 +17,17 @@ use std::convert::Infallible;
 impl<BL, BA, BI> Actions for alice::State<Ethereum, BL, asset::Erc20, BA, identity::Ethereum, BI>
 where
     BL: Ledger,
-    (BL, BA): RedeemAction<BL, BA, BI>,
+    BA: Clone,
+    BI: Clone,
+    (BL, BA): RedeemAction<HtlcParams = HtlcParams<BL, BA, BI>, HtlcLocation = BL::HtlcLocation>,
```

Comment:
This change is unfortunate, I would have loved if I could have just not specified the associated types here. Maybe there is still a way to do this but for now, this allows us to move forward!